make/src/special_target.rs

```diff
@@ -211,4 +250,80 @@ impl Processor<'_> {
 
         Ok(())
     }
+
+    fn process_suffixes(self) -> Result<(), Error> {
+        let suffixes_key = ".SUFFIXES";
+        let suffixes_set = self
+            .rule
+            .prerequisites()
+            .map(|suffix| suffix.as_ref().to_string())
+            .collect::<BTreeSet<String>>();
+
+        if suffixes_set.is_empty() {
+            self.make
+                .config
+                .rules
+                .insert(suffixes_key.to_string(), BTreeSet::new());
+        } else {
+            self.make
+                .config
+                .rules
+                .insert(suffixes_key.to_string(), suffixes_set);
+        }
+
+        Ok(())
+    }
+    fn process_phony(mut self) -> Result<(), Error> {
+        let suffixes_set = self
+            .rule
+            .prerequisites()
+            .map(|suffix| suffix.as_ref().to_string())
+            .collect::<BTreeSet<String>>();
+
+        self.make
+            .config
+            .rules
+            .insert(".PHONY".to_string(), suffixes_set);
+
+        let what_to_do = |rule: &mut Rule| rule.config.phony = true;
+        self.additive(what_to_do);
+        self.global(what_to_do);
+
+        Ok(())
+    }
+
+    fn process_precious(mut self) -> Result<(), Error> {
+        let precious_set = self
+            .rule
+            .prerequisites()
+            .map(|val| val.as_ref().to_string())
+            .collect::<BTreeSet<String>>();
+
+        let what_to_do = |rule: &mut Rule| rule.config.precious = true;
+
+        self.additive(what_to_do);
+        self.global(what_to_do);
+
+        self.make
+            .config
+            .rules
+            .insert(".PRECIOUS".to_string(), precious_set);
+        Ok(())
+    }
+    fn process_sccs_get(self) -> Result<(), Error> {
+        self.without_prerequisites()?;
+
+        let sccs_set = self
+            .rule
+            .recipes()
+            .map(|val| val.as_ref().to_string())
+            .collect::<BTreeSet<String>>();
```

Comment:
This logic is used in all the `process_` functions, it can probably be moved to the `modifiers` impl block to avoid repetitions.

Reply:
Code is substantially different to justify the duplication